tests/unit/conftest.py

```diff
@@ -683,3 +694,11 @@ def cli_args() -> MagicMock:
     """
     # spec_set needs an instantiated class to be strict with the fields.
     return MagicMock(spec_set=CLIargs(command="plan"))()
+
+
+def _generate_mock_machine(machine_id, hostname, az):
```

Comment:
Why this is "private" function? You are using it outside of conftest + this is helper function in tests, so there is not much reason to use "_".